src/models/util/LoadShow.ts

```diff
@@ -0,0 +1,56 @@
+import Show from '../Show';
+import { ReadFourCharCode, ReadTwoCharCode } from './ParseCalChartUtils';
+import { ParseCalChart } from './ParseCalChart';
+import { ParseCalChart3 } from './ParseCalChart3';
+
+// Importing calchart4 is not yet supported
+/* eslint-disable @typescript-eslint/no-unused-vars */
+function IsCalChart4(buffer: ArrayBuffer): ParseCalChart|null {
+  // console.log('no support for CalChart4.0 yet');
+  return null;
+}
+/* eslint-enable @typescript-eslint/no-unused-vars */
+
+function IsCalChart3(buffer: ArrayBuffer): ParseCalChart|null {
+  const view = new DataView(buffer, 0, 8);
+  // CalChart3 starts with INGLGUxy, where xy is used to determine the version
+  if (ReadFourCharCode(view, 0) !== 'INGL'
+      || ReadTwoCharCode(view, 4) !== 'GU') {
+    return null;
+  }
+
+  // Extract the version number.
+  const version = ReadTwoCharCode(view, 6);
+  // version "RK" is the "GURK" show, which is CalChart 3.1 - 3.3.5.
+  if (version === 'RK') {
+    // we throw here so the error gets to the user.
+    throw 'CalChart 3.3.5 and earlier not supported.';
+  }
+  // the version is actual encoded as a string, so convert to a number:
+  const versionNumber = (version.charCodeAt(0) - '0'.charCodeAt(0)) * 10
+                        + version.charCodeAt(1) - '0'.charCodeAt(0);
+  if (versionNumber < 34) {
+    throw 'CalChart 3.3.5 and earlier not supported.';
+  }
+  return new ParseCalChart3();
+}
+
+/**
+ * LoadShow
+ *
+ * @param buffer The Byte array of data to load.
+ * @returns Returns either a new [Show] or will throw an error that can be
+ * displayed to the user
+ */
+export const LoadShow = (buffer: ArrayBuffer): Show => {
+  let parser = IsCalChart4(buffer);
```

Comment:
I think it might be overkill to create classes for the parsers. Can we keep it simple and keep the parsers as their own functions? Also, can we replace `IsCalChart3` and `IsCalChart4` with a function `getShowVersion(buffer)`? It doesn't have to be exact, we just need to know if the version is in one of three zones:

1. `<=3.3.5`
2. `>3.4.0, <4`
3. `>=4`

Maybe we can create an enum for this:
```ts
enum SHOW_VERSIONS {
  THREE_OLD,
  THREE_NEW,
  FOUR,
}
```

Thus, this section of code would look like:
```ts
const showVersion: SHOW_VERSIONS = getShowVersion(buffer);
switch (getShowVersion(buffer)) {
  case SHOW_VERSIONS.THREE_OLD:
    throw 'Shows from CalChart versions older than 3.4.0 are not supported.'
  case SHOW_VERSIONS.THREE_NEW:
    return parseShowVersion3(buffer);
  case SHOW_VERSIONS.FOUR:
    return parseShowVersion4(buffer);
  default:
    throw 'Unable to read CalChart version from show file.'
}
```

Reply:
The parsers may need to have temporary data for doing consistency checks.  For example, the number of points needs to be know before we parse the labels.  And I'm anticipating that when we do translations from CalChart3.1 to CalChart4 there's going to be work when we have to go from Sheet continuity labels of the old system to the new system we're going to need to have additional temporary data that's going to have to live outside the show but in function calls.

I guess we could have them be hidden state objects behind the function call, but effectively they are the same.  So I'd rather have a factory function approach to creating objects.